### PR TITLE
UI: Namespace validation update

### PR DESCRIPTION
**Pull request by hashishaw**
The namespace path validation in the UI was out of sync with the rules described in the [documentation](https://developer.hashicorp.com/vault/docs/enterprise/namespaces#namespace-naming-restrictions). This PR updates the UI to match the guidance in the docs, and glimmerizes a couple components while we're at it.  

Validation message on namespace create form

### CODE REVIEW

File: ui/lib/core/addon/templates/components/edit-form.hbs [hunk 1]
```diff
@@ -9,17 +9,29 @@
...
     {{else if this.model.fieldGroups}}
-      <FormFieldGroups @model={{this.model}} @mode={{@mode}} />
+      <FormFieldGroups @model={{this.model}} @mode={{@mode}} @modelValidations={{this.modelValidations}} />
     {{/if}}
   </div>
   <FormSaveButtons
     @isSaving={{this.save.isRunning}}
     @saveButtonText={{this.saveButtonText}}
-    @canceLinkParams={{@canceLinkParams}}
+    @cancelLinkParams={{@cancelLinkParams}}
```
Comment (hashishaw) [inline ui/lib/core/addon/templates/components/edit-form.hbs:27]: typo from long long ago 🙃

File: ui/app/models/namespace.js [hunk 2]
```diff
@@ -5,19 +5,23 @@
...
+@withModelValidations({
+  path: [
+    { type: 'presence', message: `Path can't be blank.` },
+    { type: 'endsInSlash', message: `Path can't end in forward slash '/'.` },
+    {
+      type: 'containsWhiteSpace',
+      message: "Path can't contain whitespace.",
+    },
+  ],
+})
```
Comment (zofskeez) [inline ui/app/models/namespace.js:20]: 🎉

File: ui/lib/core/addon/components/form-field.hbs [hunk 3]
```diff
@@ -281,12 +281,6 @@
           class="input {{if this.validationError 'has-error-border'}}"
           maxLength={{@attr.options.characterLimit}}
         />
-        {{! TODO: explore removing in favor of new model validations pattern since it is only used on the namespace model }}
```
Comment (zofskeez) [inline ui/lib/core/addon/components/form-field.hbs:284]: ✅

**Review by zofskeez — APPROVED** [review]
🚀

**Pull request merged**